src/components/dashboard/WhoContent.native.js

```diff
@@ -0,0 +1,194 @@
+import React, { useCallback, useEffect } from 'react'
+import { FlatList, PermissionsAndroid } from 'react-native'
+import Contacts from 'react-native-contacts'
+import { get, orderBy, uniq } from 'lodash'
+import { isAndroid } from '../../lib/utils/platform'
+import { Section } from '../common'
+import Separator from '../common/layout/Separator'
+import InputText from '../common/form/InputText'
+import { withStyles } from '../../lib/styles'
+import normalize from '../../lib/utils/normalizeText'
+import { getDesignRelativeHeight } from '../../lib/utils/sizes'
+import userStorage from '../../lib/gundb/UserStorage'
+import FeedContactItem from './FeedContactItem'
+
+const WhoContent = ({ styles, setContact, error, text, value, next, state, showNext, setValue }) => {
+  const [contacts, setContacts] = React.useState([])
+  const [initialList, setInitalList] = React.useState(contacts)
+  const [recentFeedItems, setRecentFeedItems] = React.useState([])
+  const [recentlyUsedList, setRecentlyUsedList] = React.useState([])
+
+  const getAllContacts = () => {
+    Contacts.getAll((err, contacts) => {
+      if (err === 'denied') {
+        console.warn('permissions denied')
+      } else {
+        const sortContacts = orderBy(contacts, ['givenName'])
+        setContacts(sortContacts)
+        setInitalList(sortContacts)
+      }
+    })
+  }
+
+  const getUserFeed = async () => {
+    const userFeed = await userStorage.getFeedPage(5, true)
+    const recent = userFeed.filter(({ type }) => type === 'send' || type === 'receive')
+    setRecentFeedItems(recent)
+  }
+
+  const showPermissionsAndroid = () => {
+    PermissionsAndroid.request(PermissionsAndroid.PERMISSIONS.READ_CONTACTS, {
+      title: 'Contacts',
+      message: 'We need access to view your contacts, so you can easily send G$ to them.',
+      buttonPositive: 'Approve',
+    }).then(() => {
+      getAllContacts()
+    })
+  }
+
+  useEffect(() => {
+    getUserFeed()
+  }, [])
+
+  useEffect(() => {
+    if (Contacts) {
+      if (isAndroid) {
+        showPermissionsAndroid()
+      } else {
+        getAllContacts()
+      }
+    }
+  }, [])
+
+  const handleSearch = useCallback(query => {
+    const queryIsNumber = parseInt(query)
+    if (state) {
+      setValue(query)
+    }
+    if (query && !query.includes('+') && !query.includes('*')) {
```

Comment:
logic is too complicated
alphabet chars cannot be inside phone numbers fields (limited by the input mask)
as the opposite, numeric chars are rarely used in first/last names

so you don't need to check is the query a phone number or it looks like person's name
just search by  "phone or name contains query" condition:

```javascript
import { memoize, map } from 'lodash'

const filterContacts = useCallback(memoize(searchQuery => {
  const query = searchQuery.toLocaleLowerCase()
  
  return initialList.filter({ givenName, familyName, phoneNumbers }) => {
    const [phoneNumber] = map(phoneNumbers, 'number')

    return [givenName, familyName, phoneNumber]
      .some(field => (field || '').toLocaleLowerCase().includes(query)
  })
}, [initialList])

const handleSearch = useCallback(query => {
  if (state) {
     setValue(query)
   }

  setContacts(filterContacts(query))


}, [setContacts, setValue,  filterContacts])

Reply:
also it's better for performance to change search results via `memoize`